pkg/revisions/default_watcher.go

```diff
@@ -53,35 +49,29 @@ type defaultWatcher struct {
 	defaultRevision string
 	handlers        []DefaultHandler
 
-	queue           workqueue.RateLimitingInterface
-	webhookInformer cache.SharedInformer
-	initialSync     *atomic.Bool
+	queue           controllers.Queue
+	webhookInformer cache.SharedIndexInformer
 	mu              sync.RWMutex
 }
 
 func NewDefaultWatcher(client kube.Client, revision string) DefaultWatcher {
 	p := &defaultWatcher{
-		revision:    revision,
-		queue:       workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
-		initialSync: atomic.NewBool(false),
-		mu:          sync.RWMutex{},
+		revision: revision,
+		mu:       sync.RWMutex{},
 	}
+	p.queue = controllers.NewQueue(controllers.WithName("default revision"), controllers.WithReconciler(p.setDefault))
```

Comment:
How do we detect default webhook deletion now? Before if we got a delete event, we'd call `setDefault` with `""`, now it seems like we'd set the default to the last active default?

Reply:
```
	revision := ""
	wh, _, _ := p.webhookInformer.GetIndexer().GetByKey(key.Name)
	if wh != nil {
		revision = wh.(metav1.Object).GetLabels()[label.IoIstioRev.Name]
	}
```

if its delete `wh` is nil and therefor revision is `""`